### PR TITLE
reservation schema 

### LINKED ISSUE #3: Schema with reservations
_id - reservation id (pamiętać żeby było w mongodb)
hotelId
roomId
startDate
endDate
people: {
   adult
   children
}
createDate (pamiętać żeby było w mongodb)
updateDate (pamiętać żeby było w mongodb)
isPaid

### PR DESCRIPTION
**Pull request by hoolek77**

### CODE REVIEW

File: schemas/reservation.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  startDate: {
+    type: Date,
+    required: true,
+  },
+  endDate: {
+    type: Date,
+    required: true,
+  },
+  people: {
+    type: new mongoose.Schema({
```
Comment (bartoszbialecki) [inline schemas/reservation.js]: If so, then it should be in the separate file.

File: schemas/reservation.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
+const Joi = require('joi')
+const mongoose = require('mongoose')
+
+const reservationSchema = new mongoose.Schema({
```
Comment (pdybowski) [inline schemas/reservation.js]: We could also have resevrationId

Comment (hoolek77) [reply schemas/reservation.js]: mongo should add property _id by default

File: schemas/reservationSchema.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        min: 0,
+      },
+    },
+    required: true,
+  },
+  isPaid: {
+    type: Boolean,
+    default: false,
+  },
+  timestamps: {
```
Comment (bartoszbialecki) [inline schemas/reservationSchema.js]: I think you can do something like { useTimestamps: true }

File: schemas/reservationSchema.js [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+  startDate: {
+    type: Date,
+    required: true,
+  },
+  endDate: {
+    type: Date,
+    required: true,
+  },
+  people: {
+    type: {
```
Comment (bartoszbialecki) [inline schemas/reservationSchema.js]: is it right? I mean type should be here?

File: schemas/reservationSchema.js [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+        min: 0,
+      },
+    },
+    required: true,
+  },
+  isPaid: {
+    type: Boolean,
+    default: false,
+  },
+  timestamps: true,
```
Comment (bartoszbialecki) [inline schemas/reservationSchema.js]: did you test it?

File: schemas/reservationSchema.js (hunk 4 above)
Comment (pdybowski) [reply schemas/reservationSchema.js]: you can remove type, just pass adults and children
or create new schema 'whatevername' with adults and children, and make people: 'whatevername'

File: schemas/reservationSchema.js (hunk 5 above)
Comment (pdybowski) [reply schemas/reservationSchema.js]: it should work due to mongoose docs. We can change it later if it wont work. Its not that much important for now

Comment (bartoszbialecki) [reply schemas/reservationSchema.js]: I mean, that I saw it should be an object { timestamps: true } after these object with fields

Comment (pdybowski) [reply schemas/reservationSchema.js]: oh, yes

**Comment by pdybowski** [comment]
Raczej git

**Review by bartoszbialecki — APPROVED** [review]

**Review by ibednorz — APPROVED** [review]

**Pull request merged**